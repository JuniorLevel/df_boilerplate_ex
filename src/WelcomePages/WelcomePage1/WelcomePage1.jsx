// @flow

import {
	FourImagesBg,
	Welcome,
	WelcomeAuthButtons,
	WelcomeButton,
} from '@juniorlevel/df_boilerplate-library';
import React from 'react';

interface IWelcomePage1Props {
	imagesUrls: { id: number, url: string }[];
}

export const WelcomePage1 = ({
	imagesUrls,
}: IWelcomePage1Props): React.Node => {
	const buttons = [
		{
			id: 1,
			title: 'email',
			onClick: () => alert('email'),
		},
		{
			id: 2,
			title: 'LogIn',
			onClick: () => alert('LogIn'),
		},
		{
			id: 3,
			title: 'CreateAccount',
			onClick: () => alert('CreateAccount'),
		},
	];

	return (
		<FourImagesBg imagesUrls={imagesUrls}>
			<Welcome buttons={buttons}>
				<WelcomeAuthButtons>
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
				</WelcomeAuthButtons>
			</Welcome>
		</FourImagesBg>
	);
};
