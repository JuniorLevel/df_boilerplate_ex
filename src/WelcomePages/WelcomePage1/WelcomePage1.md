```jsx { "file": "./WelcomePage1.jsx" }

```

```jsx noeditor
import Image1 from '../welcome-img1.png';
import Image3 from '../welcome-img3.png';
<div>
	<div>
		<img src={Image1} />
	</div>
	<div>
		<img src={Image3} />
	</div>
</div>;
```
