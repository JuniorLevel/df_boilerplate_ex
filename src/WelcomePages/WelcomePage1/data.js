/* eslint-disable no-alert */
// @flow

import Image1 from '../pictures/Image1.jpg';
import Image2 from '../pictures/Image2.jpg';
import Image3 from '../pictures/Image3.jpg';
import Image4 from '../pictures/Image4.jpg';

export const images = [
	{ id: 1, url: Image1 },
	{ id: 2, url: Image2 },
	{ id: 3, url: Image3 },
	{ id: 4, url: Image4 },
];

export const welcomeButtons = [
	{
		id: 1,
		title: 'email',
		onClick: (): void => alert('email'),
	},
	{
		id: 2,
		title: 'LogIn',
		onClick: (): void => alert('LogIn'),
	},
	{
		id: 3,
		title: 'CreateAccount',
		onClick: (): void => alert('CreateAccount'),
	},
];

export const authButtons = [
	{ id: 1, title: 'G', onClick: (): void => alert('click1') },
	{ id: 2, title: 'VK', onClick: (): void => alert('click2') },
];
