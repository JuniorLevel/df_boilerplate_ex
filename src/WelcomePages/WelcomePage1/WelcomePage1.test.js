import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { WelcomePage1 } from './WelcomePage1';

describe('WelcomePage1', () => {
	test('should render WelcomePage1 component without crashing', () => {
		const { container } = render(<WelcomePage1 />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<WelcomePage1 />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<WelcomePage1 />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<WelcomePage1 />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
