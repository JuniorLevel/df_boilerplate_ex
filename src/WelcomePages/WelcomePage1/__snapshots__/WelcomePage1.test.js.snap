// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage1 matches the snapshot desktop 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1q6880g-0 dxcnEG"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-1c0na6j ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage1 matches the snapshot laptop 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1q6880g-0 dxcnEG"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-1c0na6j ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage1 matches the snapshot mobile 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1q6880g-0 dxcnEG"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-1c0na6j ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
